### PR TITLE
Move inconsistent namespace check to pre-commit, fixup more files

### PR DESCRIPTION
**Pull request by MarcoGorelli**
xref https://github.com/pandas-dev/pandas/pull/37401#discussion_r515312381

### CODE REVIEW

File: pandas/tests/indexes/period/test_formats.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def test_representation_to_series(self):
             [idx1, idx2, idx3, idx4, idx5, idx6, idx7, idx8, idx9],
             [exp1, exp2, exp3, exp4, exp5, exp6, exp7, exp8, exp9],
         ):
-            result = repr(pd.Series(idx))
+            result = repr(Series(idx))
```
Comment (MarcoGorelli) [inline pandas/tests/indexes/period/test_formats.py:157]: on line 117 we have
```
exp1 = """Series([], dtype: period[D])"""
```
so this is to be consistent with that

File: pandas/tests/indexes/timedeltas/test_formats.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def test_representation_to_series(self):
             for idx, expected in zip(
                 [idx1, idx2, idx3, idx4, idx5], [exp1, exp2, exp3, exp4, exp5]
             ):
-                result = repr(pd.Series(idx))
+                result = repr(Series(idx))
```
Comment (MarcoGorelli) [inline pandas/tests/indexes/timedeltas/test_formats.py:65]: for consistency with
```
exp1 = """Series([], dtype: timedelta64[ns])"""
```
on line 46

File: pandas/tests/resample/test_time_grouper.py [hunk 3]
```diff
@@ -306,21 +306,21 @@ def test_groupby_resample_interpolate():
...
-            (50, pd.Timestamp("2018-01-14")),
-            (50, pd.Timestamp("2018-01-15")),
-            (50, pd.Timestamp("2018-01-16")),
-            (50, pd.Timestamp("2018-01-17")),
-            (50, pd.Timestamp("2018-01-18")),
-            (50, pd.Timestamp("2018-01-19")),
-            (50, pd.Timestamp("2018-01-20")),
-            (50, pd.Timestamp("2018-01-21")),
-            (60, pd.Timestamp("2018-01-14")),
+            (50, Timestamp("2018-01-08")),
```
Comment (MarcoGorelli) [inline pandas/tests/resample/test_time_grouper.py:309]: These are for consistency with
```
        "convention='e', origin=Timestamp('2000-01-01 00:00:00'))"
```
on line 262

**Review by dsaxton — COMMENTED** [review]
Thanks for doing this @MarcoGorelli!

Is the regex we're using correct here? When I try to make a "bad" commit (adding pd to Categorical unnecessarily) locally it seems I'm able to:

```
(pandas-dev) ~/pandas (inconsistent-namespace) $ git commit -m "Should fail"
black..................................................................................................Passed
flake8.................................................................................................Passed
flake8 (cython)....................................................................(no files to check)Skipped
flake8 (cython template)...........................................................(no files to check)Skipped
isort (python).........................................................................................Passed
isort (cython).....................................................................(no files to check)Skipped
pyupgrade..............................................................................................Passed
rst ``code`` is two backticks......................................................(no files to check)Skipped
rst directives end with two colons.....................................................................Passed
rst ``inline code`` next to normal text................................................................Passed
Generate pip dependency from conda.................................................(no files to check)Skipped
flake8-rst.........................................................................(no files to check)Skipped
Check for non-standard imports.........................................................................Passed
Check for non-standard numpy.random-related imports excluding pandas/_testing.py.......................Passed
Check for non-standard imports in test suite...........................................................Passed
Check for incorrect code block or IPython directives...................................................Passed
Check for use of not concatenated strings..............................................................Passed
Check for strings with wrong placed spaces.............................................................Passed
Check for import of private attributes across modules..............................(no files to check)Skipped
Check for use of private functions across modules..................................(no files to check)Skipped
Check for inconsistent use of pandas namespace in tests................................................Passed
Check for use of Union[Series, DataFrame] instead of FrameOrSeriesUnion alias..........................Passed
Check for use of foo.__class__ instead of type(foo)....................................................Passed
Check for use of comment-based annotation syntax and missing error codes...............................Passed
Check code for instances of os.remove..................................................................Passed
Strip unnecessary `# noqa`s............................................................................Passed
Fix End of Files...................................................................(no files to check)Skipped
Trim Trailing Whitespace...............................................................................Passed
[inconsistent-namespace 4ef64f857] Should fail
 1 file changed, 1 insertion(+), 1 deletion(-)
```

```
(pandas-dev) ~/pandas (inconsistent-namespace) $ git show
commit 4ef64f857b681cf7e4211faaec4c7dd2d2b27be5 (HEAD -> inconsistent-namespace)
Author: Daniel Saxton <dsaxton@pm.me>
Date:   Fri Nov 6 08:19:00 2020 -0600

    Should fail

diff --git a/pandas/tests/groupby/transform/test_transform.py b/pandas/tests/groupby/transform/test_transform.py
index d7426a5e3..e96d63f9d 100644
--- a/pandas/tests/groupby/transform/test_transform.py
+++ b/pandas/tests/groupby/transform/test_transform.py
@@ -968,7 +968,7 @@ def test_transform_absent_categories(func):
     x_vals = [1]
     x_cats = range(2)
     y = [1]
-    df = DataFrame(dict(x=Categorical(x_vals, x_cats), y=y))
+    df = DataFrame(dict(x=pd.Categorical(x_vals, x_cats), y=y))
     result = getattr(df.y.groupby(df.x), func)()
     expected = df.y
```

**Comment by MarcoGorelli** [comment]
Good catch, this one isn't picked up because we have the first instance of `Categorical` in that file doesn't have an open parenthesis after it. Have updated (with a little test too)

**Review by dsaxton — APPROVED** [review]
Thanks again @MarcoGorelli

**Comment by jreback** [comment]
ok this looks good, can you merge master one more time and ping on green.

**Comment by jbrockmendel** [comment]
restarted the docbuild, which looked like an HTTPError

**Comment by MarcoGorelli** [comment]
Thanks - @jreback green now

**Pull request merged**